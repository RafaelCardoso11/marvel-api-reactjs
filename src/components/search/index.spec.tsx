import { render } from "@testing-library/react";
import { Search } from "./index";

describe("<Search/>", () => {
  it("Deveria renderizar na tela o componente Search com um Input", () => {
    const setValueAfterWrite = jest.fn();
    const screen = render(<Search setValueAfterWrite={setValueAfterWrite} />);

    const searchInput = screen.getByTestId("id-search-characters");
    expect(searchInput).toBeInTheDocument();

    expect(screen.container).toMatchSnapshot();
  });
  it.todo("Deveria ser possível buscar");
  it.todo("Deveria ser possível buscar de forma Debounce");
});
 