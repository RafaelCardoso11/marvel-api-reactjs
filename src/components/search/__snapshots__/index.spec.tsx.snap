// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> Deveria renderizar na tela o componente Search com um Input 1`] = `
<div>
  <input
    class="sc-kvAqsT"
    data-testid="id-search-characters"
    placeholder="Pesquisar..."
    type="text"
    value=""
  />
</div>
`;
